### PR TITLE
FEATURE: match user title when primary group changes

### PR DESCRIPTION
**Pull request by xrav3nz**
> DISCUSSION: https://meta.discourse.org/t/user-in-multiple-groups-title-display-and-group-link-dont-match/96502/17?u=xrav3nz

When primary group changes and the user's title is the previous primary group's title, change the title to the new primary group's title

---

- [x] More specs on existing `Group` behavior (https://github.com/discourse/discourse/pull/6403)
- [x] Switch title if exact match and primary changes (this PR)
- [ ] When stripping a title, we should check if there is another title we can grant the user

### CODE REVIEW

**Comment by SamSaffron** [comment]
Ideally let's keep "work in progress" PRs in your own branch, you can always link me to it 😄

**Comment by xrav3nz** [comment]
I actually meant to split it down into separate PRs (thus the three checkboxes linking to different PRs). But for sure I will keep that in mind.

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -1814,4 +1814,29 @@ def filter_by(method)
...
+    let(:new_primary_group) { Fabricate(:group, title: 'New', users: [user]) }
+
+    before { user.update(primary_group: previous_primary_group) }
+
+    def change_primary_group
+      user.update(primary_group: new_primary_group)
+    end
+
+    it "updates user's title when it matches the previous primary group" do
+      user.update(title: 'Previous')
```
Comment (SamSaffron) [inline spec/models/user_spec.rb]: lets merge all these 3 specs if we can cleanly? Overall the change looks good to me. 👍

**Comment by SamSaffron** [comment]
looks good to me!

**Pull request merged**